tracer/src/Datadog.Trace/Configuration/ImmutableTracerSettings.cs

```diff
@@ -528,6 +530,11 @@ internal ImmutableTracerSettings(TracerSettings settings, bool unusedParamNotToU
         /// </summary>
         internal bool IsDataStreamsMonitoringEnabled => DynamicSettings.DataStreamsMonitoringEnabled ?? _isDataStreamsMonitoringEnabled;
 
+        /// <summary>
+        /// Gets a value indicating whether data streams monitoring legacy headers are enabled or not.
+        /// </summary>
+        internal bool IsDataStreamsLegacyHeadersEnabled => DynamicSettings.DataStreamsLegacyHeadersEnabled ?? _isDataStreamsLegacyHeadersEnabled;
```

Comment:
If you remove this from `DynamicSettings` (see my other comment), then you can remove the `_isDataStreamsLegacyHeadersEnabled` and convert `IsDataStreamsLegacyHeadersEnabled` into a get-only property (like the ones with `{ get; }` below).